packages/cli/cloud/src/deploy-project/action.ts

```diff
@@ -214,6 +251,21 @@ export default async (ctx: CLIContext) => {
     maxSize = 100000000;
   }
 
+  let targetEnvironment: string;
+
+  if (opts.environment) {
+    if (!environments.includes(opts.environment)) {
+      ctx.logger.error(`Environment ${opts.environment} does not exist.`);
```

Comment:
Maybe we should warn them about not being able to match the environment and proceed to the promptForEnvironment func? 🤔 
Although not sure it would please ppl wanting to use this in CI, but I doubt we have many CI users

Reply:
The idea was that the user can use the flag (opts) to automate the deployment to an environment. Adding a step that requires human intervention defeats this purpose. We could solve that with a timeout, but I would rather make the "--flag way" as straightforward as possible: IMO the user is supposed to know what they are doing when using flags, if they pass the wrong env value, it should simply fail. The interactive path should be reserved for the regular way of using the command.

Reply:
I'm ok with this approach 👍